src/frontends/onnx/frontend/src/op/com.microsoft/pad.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "op/com.microsoft/pad.hpp"
+
+#include "core/null_node.hpp"
+#include "exceptions.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/pad.hpp"
+#include "openvino/op/squeeze.hpp"
+#include "openvino/op/util/op_types.hpp"
+#include "utils/reshape.hpp"
+#include "utils/split.hpp"
+
+namespace {
+ov::op::PadMode get_pad_mode(std::string mode) {
+    ov::op::PadMode pad_mode;
+
+    if (mode == "constant") {
+        pad_mode = ov::op::PadMode::CONSTANT;
+    } else if (mode == "reflect") {
+        pad_mode = ov::op::PadMode::REFLECT;
+    } else if (mode == "edge") {
+        pad_mode = ov::op::PadMode::EDGE;
+    } else {
+        OPENVINO_THROW("Unsupported padding mode: [" + mode + "]");
+    }
+
+    return pad_mode;
+}
+}  // namespace
+
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace onnx {
+namespace op {
+namespace custom {
+namespace set_1 {
+ov::OutputVector pad(const ov::frontend::onnx::Node& node) {
+    const auto inputs = node.get_ov_inputs();
+    const auto& data = inputs[0];
+    const auto& pads_input = inputs[1];
+    auto pads = pads_input;
+    if (pads.get_shape().size() == 2) {
+        pads = std::make_shared<v0::Squeeze>(pads);
+    }
```

Comment:
As I mentioned, `get_shape()` called on the node with dynamic shape will cause an issue, so it should be ensured that the shape is static before the rank size check.

```suggestion
    const auto& pads_pshape = pads.get_partial_shape();
    if (pads_pshape.rank().is_static() && pads_pshape.size() == 2) {
        pads = std::make_shared<v0::Squeeze>(pads);
    }
```